chain/actors/builtin/power/diff.go

```diff
@@ -0,0 +1,117 @@
+package power
+
+import (
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	cbg "github.com/whyrusleeping/cbor-gen"
+
+	"github.com/filecoin-project/lotus/chain/actors/adt"
+)
+
+type ClaimChanges struct {
+	Added    []ClaimInfo
+	Modified []ClaimModification
+	Removed  []ClaimInfo
+}
+
+type ClaimModification struct {
+	Miner address.Address
+	From  Claim
+	To    Claim
+}
+
+type ClaimInfo struct {
+	Miner address.Address
+	Claim Claim
+}
+
+func DiffClaims(pre, cur State) (*ClaimChanges, error) {
```

Comment:
@Stebalien would you mind giving this a second look, I've added this since the initial request for review.